### PR TITLE
Reverts https://github.com/crowbar/crowbar-core/pull/265 and refactors code

### PR DESCRIPTION
**Pull request by sayalilunkad**

### CODE REVIEW

File: chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb [hunk 1]
```diff
@@ -127,13 +103,16 @@
...
 
-  # Stop other services
+  # Stop the database and corosync
   service "drbd" do
     action :stop
   end
   service "openais" do
     action :stop
   end
+  service "postgresql" do
```
Comment (jsuchome) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: Shouldn't this be before drbd?

Comment (sayalilunkad) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: Not sure, I put it in the same order as before.

Comment (jsuchome) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: True.

**Comment by jsuchome** [comment]
+1

**Comment by aplanas** [comment]
+1

File: chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb [hunk 2]
```diff
@@ -33,17 +33,14 @@
...
-    EOF
+  service "drbd" do
+    action :stop
+  end
+  service "openais" do
+    action :stop
+  end
+  service "openais-shutdown" do
+    action :stop
   end
```
Comment (vuntz) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: Hrm, does that really work if the services don't exist?

Comment (vuntz) [inline chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb]: (I guess it works, though, I'm just surprised :-))

**Pull request merged**